spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md

```diff
@@ -37,77 +40,126 @@ spring.cloud.azure.appconfiguration.fail-fast | Whether throw RuntimeException o
 
-In this library, managed service identity is used to retrieve the connection string of the configuration store, the connection string is not required if running the Spring Boot application on Azure with managed service identity enabled.
+In this library, [Azure Identity SDK][azure_identity_sdk] is used to access Azure App Configuration and optionally Azure Key Vault, for secrets. The connection string is not required and will be ignored if Managed Identity is being used.
 
 Follow below steps to enable managed service identity feature:
 
-1. [Enable managed identities service](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-can-i-use-managed-identities-for-azure-resources) for virtual machine or App Service, on which the application will be deployed
+1. [Enable managed identities service](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview#how-can-i-use-managed-identities-for-azure-resources) for virtual machine or App Service, on which the application will be deployed
+
+1. Configure the [Azure RBAC](https://docs.microsoft.com/azure/role-based-access-control/role-assignments-portal) to allow application running on VM or App Service to access the configuration store. When prompted for the Role select App Configuration Data Reader, App Configuration Data Owner is not required but can be used if needed.
+
+1. Choose Configuration option:
+    1. Environment variables, only supports System Assigned Managed Identity and Service Principle.
+    1. Create a TokenCredentialProvider and supply any valid TokenCredential and supply it via a Bean.
+    1. Configure bootstrap.properties(or .yaml) in the Spring Boot application.
+
+The configuration store name must be configured when `connection-string` is empty, the connection string for the configuration store will be loaded automatically.
+
+### Token Credential Provider
 
-2. Configure the [Azure RBAC](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal) to allow application running on VM or App Service to access the configuration store. When propted for the Role select Contributor.
- 
-3. Configure bootstrap.properties(or .yaml) in the Spring Boot application as following:
+```java
+public class MyCredentials implements TokenCredentialProvider {
+
+    @Override
+    public TokenCredential credentialForAppConfig() {
+            return buildCredential();
+    }
+
+    @Override
+    public TokenCredential credentialForKeyVault() {
+            return buildCredential();
+    }
+
+    TokenCredential buildCredential() {
+            return new ManagedIdentityCredentialBuilder()
+                    .clientId("bf2043ad-c58a-4800-ad35-9a7bc8ddfdea")
+                    .build();
+    }
+
+}
 ```
+
+### bootstrap.application
+
+```application
 spring.cloud.azure.appconfiguration.stores[0].name=[config-store-name]
```

Comment:
I think we will want to deprecate this and replace it with `spring.cloud.azure.appconfiguration.stores[0].endpoint`